### PR TITLE
transact basic transfer

### PR DESCRIPTION
**Pull request by hotay**
https://github.com/systemaccounting/mxfactorial/issues/28

### CODE REVIEW

**Comment by hotay** [comment]
@mxfactorial I finished this, please help to test

**Comment by elvongray** [comment]
Hi @hotay, I was asked by @mxfactorial to review this pull request.

File: app/reducers/transaction_item.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  [UPDATE_TRANSACTION]: (state, action) => {
+    const transaction_item = cloneTransactionItem(state);
+    let {key, field, value} = action.payload;
+    if (field==='quantity' || field === 'value') {
+      value = value && Number(value) || 0; 
+    }
+    transaction_item[key][field] = value;
+    return transaction_item;
+  }
+}, []);
```
Comment (elvongray) [inline app/reducers/transaction_item.js:28]: @hotay Nice implementation, but any reason why you chose to use redux-actions? I think the normal basic redux actions is sufficient for this project.

File: app/selectors/transaction.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { reduce } from 'lodash';
+import { createSelector } from 'reselect';
+
+const getTransactionItem = (state) => (state.transaction_item);
+
+export const transactionAmountSelector = createSelector(getTransactionItem, (items) => (
+  reduce(items, (memo, item) =>(memo + item.value * item.quantity), 0)
+));
```
Comment (elvongray) [inline app/selectors/transaction.js]: @hotay Also there is not much benefit in using reselect as the state tree is not large. Please try to use the basic stuffs in react and redux, we want to ensure we don't have loads of dependencies and also make it easier for other developers to work on the project.

File: app/reducers/transaction_item.js (hunk 1 above)
Comment (hotay) [inline app/reducers/transaction_item.js:28]: Redux-actions prevents boiler plate for action and provide really nice way to create action creators and reducers. Also we need our action to follow FSA, so it is the recommended way to implement.

File: app/selectors/transaction.js (hunk 2 above)
Comment (hotay) [inline app/selectors/transaction.js]: @elvongray yes, so I will remove reselect but still keep the selector, is it ok?

**Comment by mxfactorial** [comment]
Passes testing. 
Awaiting response from @elvongray to @hotay's question.

File: app/selectors/transaction.js (hunk 2 above)
Comment (elvongray) [inline app/selectors/transaction.js]: @hotay Yes it is okay

File: app/reducers/transaction_item.js (hunk 1 above)
Comment (elvongray) [inline app/reducers/transaction_item.js:28]: Alright

**Pull request merged**